CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(gsimon
+        VERSION 1.0.0
+        LANGUAGES Fortran)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+set(CMAKE_DIRECTORY_LABELS ${PROJECT_NAME})
+
+include(GNUInstallDirs)
+
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+if(NOT CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU|Intel)$")
+  message(WARNING "${CMAKE_Fortran_COMPILER_ID} is not supported.")
+endif()
+
+# Build options
+option(BUILD_UTIL_ALLMON "Build all monitoring utilities" OFF)
+option(BUILD_UTIL_MINMON "Build Minimization Monitoring Utilities" OFF)
```

Comment:
I don't think there's any harm in having a build option for the MinMon, but it doesn't actually have any executables.

Reply:
Correct.  As discussed offline, when/if this becomes an installable package, this file will come to action.